tests/qos/qos_sai_base.py

```diff
@@ -369,46 +409,79 @@ def dutConfig(self, request, duthosts, rand_one_dut_hostname, tbinfo):
-                    test ports
+                dutConfig (dict): Map of DUT config containing dut interfaces,
+                test port IDs, test port IPs, and test ports
         """
         duthost = duthosts[rand_one_dut_hostname]
+        dut_asic = duthost.asic_instance(enum_frontend_asic_index)
         dutLagInterfaces = []
-        mgFacts = duthost.get_extended_minigraph_facts(tbinfo)
-
-        for _, lag in mgFacts["minigraph_portchannels"].items():
-            for intf in lag["members"]:
-                dutLagInterfaces.append(mgFacts["minigraph_ptf_indices"][intf])
+        dutPortIps = {}
+        testPortIps = {}
 
-        testPortIds = set(mgFacts["minigraph_ptf_indices"][port] for port in mgFacts["minigraph_ports"].keys())
-        testPortIds -= set(dutLagInterfaces)
-        if isMellanoxDevice(duthost):
-            # The last port is used for up link from DUT switch
-            testPortIds -= {len(mgFacts["minigraph_ptf_indices"]) - 1}
-        testPortIds = sorted(testPortIds)
+        mgFacts = duthost.get_extended_minigraph_facts(tbinfo)
+        topo = tbinfo["topo"]["name"]
 
-        # get current DUT port IPs
-        dutPortIps = {}
-        for portConfig in mgFacts["minigraph_interfaces"]:
-            if ipaddress.ip_interface(portConfig['peer_addr']).ip.version == 4:
-                portIndex = mgFacts["minigraph_ptf_indices"][portConfig["attachto"]]
-                if portIndex in testPortIds:
-                    dutPortIps.update({portIndex: portConfig["peer_addr"]})
+        testPortIds = []
+        if topo in self.SUPPORTED_T0_TOPOS:
+            pytest_assert(
+                not duthost.sonichost.is_multi_asic, "Fixture not supported on T0 multi ASIC"
+            )
+            for _, lag in mgFacts["minigraph_portchannels"].items():
+                for intf in lag["members"]:
+                    dutLagInterfaces.append(mgFacts["minigraph_ptf_indices"][intf])
+
+            testPortIds = set(mgFacts["minigraph_ptf_indices"][port]
+                                for port in mgFacts["minigraph_ports"].keys())
+            testPortIds -= set(dutLagInterfaces)
+            if isMellanoxDevice(duthost):
+                # The last port is used for up link from DUT switch
+                testPortIds -= {len(mgFacts["minigraph_ptf_indices"]) - 1}
+            testPortIds = sorted(testPortIds)
+
+            # get current DUT port IPs
+            dutPortIps = {}
+            for portConfig in mgFacts["minigraph_interfaces"]:
+                if ipaddress.ip_interface(portConfig['peer_addr']).ip.version == 4:
+                    portIndex = mgFacts["minigraph_ptf_indices"][portConfig["attachto"]]
+                    if portIndex in testPortIds:
+                        dutPortIps.update({portIndex: portConfig["peer_addr"]})
+
+            testPortIps = self.__assignTestPortIps(mgFacts)
+
+        elif topo in self.SUPPORTED_T1_TOPOS:
```

Comment:
This is going to cause regressions on existing runs on T1 topology for single asic. Most of our duts in 't1-lag' have non Portchannel interfaces as well. We make use of those interfaces for our test currently or maybe modify the logic to check for topo and single asic and keep existing behavior

Reply:
It should work for T1 topology too, and it gets only the working IP interfaces. I'll verify it on T1 topology and update.

Reply:
verified the API works fine with Ethernet interfaces. This works fine.